Defs/Ammo/Rocket/84x246mmR.xml

```diff
@@ -42,44 +50,173 @@
+			<Mass>4.2</Mass>
+			<Bulk>10.25</Bulk>
+		</statBases>
+		<ammoClass>RocketThermobaric</ammoClass>
+		<detonateProjectile>Bullet_84x246mmR_Thermobaric</detonateProjectile>
+	</ThingDef>
+
 	<ThingDef Class="CombatExtended.AmmoDef" ParentName="84x246mmRBase">
 		<defName>Ammo_84x246mmR_HE</defName>
 		<label>84x246mmR cartridge (HE)</label>
 		<graphicData>
-			<texPath>Things/Ammo/Cannon/HE</texPath>
-			<graphicClass>Graphic_Single</graphicClass>
+			<texPath>Things/Ammo/Recoilless/HE</texPath>
+			<graphicClass>Graphic_StackCount</graphicClass>
 		</graphicData>
 		<statBases>
-			<MarketValue>54.36</MarketValue>
-			<Mass>3.1</Mass>
+			<MarketValue>78.62</MarketValue>
+			<Mass>4.3</Mass>
 			<Bulk>7.26</Bulk>
 		</statBases>
 		<ammoClass>GrenadeHE</ammoClass>
 		<detonateProjectile>Bullet_84x246mmR_HE</detonateProjectile>
 	</ThingDef>
 
+	<ThingDef Class="CombatExtended.AmmoDef" ParentName="84x246mmRBase">
+		<defName>Ammo_84x246mmR_HE_TFuzed</defName>
+		<label>84x246mmR cartridge (HE Time-Fuzed)</label>
+		<graphicData>
+			<texPath>Things/Ammo/Recoilless/HE-AB</texPath>
+			<graphicClass>Graphic_StackCount</graphicClass>
+		</graphicData>
+		<statBases>
+			<MarketValue>78.62</MarketValue>
+			<Mass>4.3</Mass>
+			<Bulk>7.26</Bulk>
+		</statBases>
+		<ammoClass>GrenadeHETF</ammoClass>
+		<detonateProjectile>Bullet_84x246mmR_HE_TFuzed</detonateProjectile>
+	</ThingDef>
+
+	<ThingDef Class="CombatExtended.AmmoDef" ParentName="84x246mmRBase">
+		<defName>Ammo_84x246mmR_HEDP</defName>
+		<label>84x246mmR cartridge (HEDP)</label>
+		<graphicData>
+			<texPath>Things/Ammo/Recoilless/HEDP</texPath>
+			<graphicClass>Graphic_StackCount</graphicClass>
+		</graphicData>
+		<statBases>
+			<MarketValue>49.68</MarketValue>
+			<Mass>3.3</Mass>
+			<Bulk>8.35</Bulk>
+		</statBases>
+		<ammoClass>GrenadeHE</ammoClass>
+		<detonateProjectile>Bullet_84x246mmR_HEDP</detonateProjectile>
+	</ThingDef>
+
+	<ThingDef Class="CombatExtended.AmmoDef" ParentName="84x246mmRBase">
+		<defName>Ammo_84x246mmR_Frag</defName>
+		<label>84x246mmR cartridge (Frag)</label>
+		<graphicData>
+			<texPath>Things/Ammo/Recoilless/FRAG</texPath>
+			<graphicClass>Graphic_StackCount</graphicClass>
+		</graphicData>
+		<statBases>
+			<MarketValue>48.89</MarketValue>
+			<Mass>3.2</Mass>
+			<Bulk>7.18</Bulk>
+		</statBases>
+		<ammoClass>RocketFrag</ammoClass>
+		<detonateProjectile>Bullet_84x246mmR_Frag</detonateProjectile>
+	</ThingDef>
+
+	<ThingDef Class="CombatExtended.AmmoDef" ParentName="84x246mmRBase">
+		<defName>Ammo_84x246mmR_CAN</defName>
+		<label>84x246mmR cartridge (CAN)</label>
+		<graphicData>
+			<texPath>Things/Ammo/Recoilless/CANISTER</texPath>
+			<graphicClass>Graphic_StackCount</graphicClass>
+		</graphicData>
+		<statBases>
+			<MarketValue>24.41</MarketValue>
+			<Mass>2.55</Mass>
+			<Bulk>7.31</Bulk>
+		</statBases>
+		<ammoClass>BuckShot</ammoClass>
+		<detonateProjectile>Bullet_84x246mmR_CAN</detonateProjectile>
```

Comment:
I'd switch this to HE or something--a stack of these cooking off is going to spawn 2k+ fragments and brick your PC.